### PR TITLE
fixed errors with user input (invest & calculator)

### PR DESCRIPTION
**Pull request by gabyteera**

### CODE REVIEW

File: src/components/investModal/index.tsx [hunk 1]
```diff
@@ -244,20 +258,29 @@ const InvestModal = ({
...
+                    BigNumber.from(ethers.utils.parseEther(value)).lt(
+                      hardCap.sub(totalInvested)
+                    ) || "Unable to invest above Hard Cap",
                   belowBalance: (value) =>
-                    value < balance ||
-                    value === balance ||
+                    BigNumber.from(ethers.utils.parseEther(value)).lt(
+                      BigNumber.from(ethers.utils.parseEther(balance))
+                    ) ||
+                    BigNumber.from(ethers.utils.parseEther(value)).eq(
```
Comment (BenasVolkovas) [inline src/components/investModal/index.tsx:276]: You can use the function `lte()` to check if the value is less than or equal to another value.

File: src/components/calculator/index.tsx [hunk 2]
```diff
@@ -127,19 +142,27 @@ const Calculator = () => {
...
+              autoComplete="off"
               name="amount"
               placeholder=""
-              defaultValue={0}
+              defaultValue={"0"}
               value={amount}
-              onChange={(e) => setAmount(e.target.valueAsNumber)}
+              onChange={(e) => setAmount(e.target.value)}
+              onKeyPress={(e) => {
+                if (
```
Comment (GabijaBeres) [inline src/components/calculator/index.tsx:152]: butu geriau sita isskelti , prilyginti variable ir cia ideti variable, reikia kuo labiau vengti kazkokios logikos jsx'e

**Review by BenasVolkovas — APPROVED** [review]

File: src/web3/getTokenBalance.ts [hunk 3]
```diff
@@ -11,22 +11,26 @@ import ERC20TokenABI from './abi/ERC20Token.json'
...
+  tokenAddress: string,
+  provider: any,
+  address?: string | null | undefined
+) => {
+  try {
+    const contract = new ethers.Contract(tokenAddress, ERC20TokenABI, provider);
+    //For testing purposes addresses that have USDC/USDT currencies in Mainnet/Goerli are hardcoded
+    //If the address from investModal is provided as an argument - balance of currently connected account will be checked
+    const tokenBalance = await contract.balanceOf(
+      address ? address : "0x618ada3f9f7BC1B2f2765Ba1728BEc5057B3DE40"
```
Comment (GabijaBeres) [inline src/web3/getTokenBalance.ts:24]: why its hardcoded?

Comment (GabijaBeres) [reply src/web3/getTokenBalance.ts:24]: Tiksliau kodel cia irasomas adresas, o ne kazkur kitur apibreztas :D

**Pull request merged**